components/pattern/src/multi_named.rs

```diff
@@ -434,38 +434,31 @@ impl<'a> MultiNamedPlaceholderPatternIterator<'a> {
         match self.store.find(|x| (x as usize) <= 0x07) {
             Some(0) => {
                 // Placeholder
-                let Some(&[lead, trail]) = self.store.as_bytes().get(0..2) else {
+                let Some((&[lead, trail], remainder)) = self
+                    .store
+                    .split_at_checked(2)
+                    .map(|(a, b)| (a.as_bytes(), b))
+                else {
                     return Err(MultiNamedPlaceholderError::InvalidStore);
                 };
                 debug_assert!(lead <= 7);
                 if trail > 7 {
                     return Err(MultiNamedPlaceholderError::InvalidStore);
                 }
                 let placeholder_len = (lead << 3) + trail;
-                let boundary = (placeholder_len as usize) + 2;
-                // TODO: use .split_at_checked() when available:
-                // https://github.com/rust-lang/rust/issues/119128
-                let Some(placeholder_name) = self.store.get(2..boundary) else {
+                let Some((placeholder_name, remainder)) =
+                    remainder.split_at_checked(placeholder_len as usize)
+                else {
                     return Err(MultiNamedPlaceholderError::InvalidStore);
                 };
```

Comment:
```suggestion
                let (placeholder_name, remainder) =
                    remainder.split_at_checked(placeholder_len as usize)
                    .ok_or(MultiNamedPlaceholderError::InvalidStore)?;
```

Reply:
The let-else is consistent with the rest of the code in this file.